fe/fe-core/src/main/java/org/apache/doris/catalog/ColumnType.java

```diff
@@ -208,12 +210,48 @@ public static void checkSupportSchemaChangeForComplexType(Type checkType, Type o
         if (checkType.isStructType() && other.isStructType()) {
```

Comment:
maybe also check `not nullable` -> `nullable` and `nullable` -> `not nullable`,  if already checked in some where else ,then add such test cases